docs/static/running-logstash-windows.asciidoc

```diff
@@ -1,3 +1,147 @@
++
+[source,txt]
+-----
+PS C:\Windows\system32> cd C:\logstash-7.1.0\
+PS C:\logstash-7.1.0> .\bin\nssm.exe install logstash
+-----
+
+Once the `NSSM service installer` window appears, specify the following parameters in the `Application` tab:
+
+** In the `Application` tab:
+*** Path: Path to `logstash.bat`: `C:\logstash-7.1.0\bin\logstash.bat`
+*** Startup Directory: Path to the `bin` directory: `C:\logstash-7.1.0\bin`
+*** Arguments: For our example to start Logstash: `-f C:\logstash-7.1.0\config\syslog.conf`
+
+** Review and make any changes necessary in the `Details` tab:
+*** Ensure `Startup Type` is set appropriately
+*** Set the `Display name` and `Description` fields to something relevant
+
+** Review any other required settings (for the example we aren't making any other changes)
+*** Be sure to determine if you need to set the `Log on` user
+** Validate the `Service name` is set appropriately
+*** For this example, we will set ours to `logstash-syslog`
+
+** Click `Install Service`
+*** Click 'OK' when the `Service "logstash-syslog" installed successfully!` window appears
+
+Once the service has been installed with NSSM, validate and start the service following the https://docs.microsoft.com/en-us/powershell/scripting/samples/managing-services[PowerShell Managing Services] documentation.
+
+[[running-logstash-windows-scheduledtask]]
+==== Running Logstash with Task Scheduler
+NOTE: It is recommended to validate your configuration works by running Logstash manually before you proceed.
+
+Open the Windows https://docs.microsoft.com/en-us/windows/desktop/taskschd/task-scheduler-start-page[Task Scheduler], then click `Create Task` in the Actions window.  Specify the following parameters in the `Actions` tab:
+
+** In the `Actions` tab:
+*** Click `New`, then specify the following:
+*** Action: `Start a program`
+*** Program/script: `C:\logstash-7.1.0\bin\logstash.bat`
+*** Add arguments: `-f C:\logstash-7.1.0\config\syslog.conf`
+*** Start in: C:\logstash-7.1.0\bin\
+
+** Review and make any changes necessary in the `General`, `Triggers`, `Conditions`, and `Settings` tabs.
+
+** Click `OK` to finish creating the scheduled task.
+
+** Once the new task has been created, either wait for it to run on the schedule or select the service then click `Run` to start the task.
+
+NOTE: Logstash can be stopped by selecting the service, then clicking `End` in the Task Scheduler window.
+
+[[running-logstash-windows-example]]
+==== Example Logstash Configuration
+We will configure Logstash to listen for syslog messages over port 514 with this configuration (file name is `syslog.conf`):
+[source,txt]
+-----
+# Sample Logstash configuration for receiving
+# UDP syslog messages over port 514
+
+input {
+  udp {
+    port => 514
+    type => "syslog"
+  }
+}
+
+output {
+  elasticsearch { hosts => ["localhost:9200"] }
+  stdout { codec => rubydebug }
+}
+-----
```

Comment:
@rwaight Can add an example of pipelines.yml down here if you want, or up above. Wherever makes sense. Monolithic pipelines have drawbacks, so some mentioning the support for multi pipelines could prove to be useful in these docs.
```
- pipeline.id: winlogbeat
  pipeline.workers: 1
  path.config: "/Program Files/ELK/logstash-6.3.2/config.d/winlogbeat.conf"
- pipeline.id: metricbeat
  pipeline.workers: 1
  path.config: "/Program Files/ELK/logstash-6.3.2/config.d/metricbeat.conf"
```

Reply:
@triggerman86, I've added a note for multiple pipeline support in [a3a75fe](https://github.com/elastic/logstash/commit/a3a75fe84300a0b39741ba1cf308dfb006a13db4)

Reply:
@triggerman86, thank you for the feedback!  You bring up a great point that this may cause more issues, especially if arguments are used.  I've made another update to clarify that in a production environment, we recommend using the logstash settings file to control Logstash execution (i.e., no `-f .\config\syslog.conf` arguments).  I added that as a note in each place there was an example of an argument being used.